### DISCUSSION TITLE
Global code indexing in Modrinth

### DISCUSSION THREAD

**Discussion by Earthcomputer** [modrinth/.github#80] [Feature requests]
**Is your feature request related to a problem? Please describe.**
Currently, it is very hard for mod developers to know whether their mod has a `@Redirect` conflict with another mod until they have released it, and receive a bug report from a user about it.
It would also be nice to get code examples from mods that call into some API.

**Describe the solution you'd like**
Wait, what does this have to do with Modrinth? Well, Modrinth hosts a large sample of mods, and an increasing number of mods are being published on Modrinth. This places Modrinth in a good position to create code indexes over many mods, and provide a web API for accessing the index.

Clients, such as the Minecraft Development plugin for IntelliJ, could use this API for the features described in the problem. For example, to find out whether the mod you're currently writing has a redirect conflict with an existing mod, mcdev just has to query the Modrinth API asking for all references to the method call you're redirecting, that are of type `Redirect`.

**Describe alternatives you've considered**
MinecraftDev could host its own web server for the same purpose. But we don't have access to as large a pool of mods. We could use modpacks, but they would get out of date quickly.

**Additional context**
I am interested in implementing this myself, but I may need a bit of guidance from a Labrinth developer, as this type of project is a bit out of my comfort zone. Please contact me here or on Discord about it.

**Comment by triphora** [comment]
The easiest place to obtain guidance in implementing this is definitely [our Discord](https://discord.gg/EUHuJHt), we'd be more than happy to help you out in the `#backend` channel.
